### PR TITLE
More immersive caves

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Makes caves more spooky.

req beacons no longer function in deep caves.
Comms works poorly in caves, and not at all in deep caves.

Grayson will review map areas after the PR is merged to make sure shallow/deep caves are actually used as you'd expect.
## Why It's Good For The Game
Caves right now are pretty cozy for marines, all their stuff works in there fine other than fire support, so they can merrily charge in and smash silos.

Caves should be the spooky xeno lair that you enter at your own peril, especially since we are now based around nuke war instead of distress signal, so entering caves is often not required, or you don't need to enter that far.

This makes caves very much the xeno home territory, generating more FUN, and making silo pushes somewhat more dangerous.
## Changelog
:cl:
balance: Comms and req no longer function in deep caves
/:cl:

### CODE REVIEW

**Comment by Losenis** [comment]
>Comms no longer work in caves
>Translation: Bald command can no longer try to have the silopushers to suicide at disks
>Because no ability to actually get them to go back more people will go into caves to try salvage the situation making the silopush even stronger
>Meanwhile engis cade up the entrances to caves for req unintentionally also locking the xenos inside caves in the chokes instead of the wide fucking open

wtf based speedmerge

**Comment by Strayed-Raven** [comment]
So I am kinda curious where the need to do this is justified. Marines already have no CAS or OB in caves. Xenos have the best position in these areas for mazes. And CQC areas by themselves in general is bias'd towards Xenos. So unless this is just degrees by degrees doctrine to be vindictive towards marines I have to ask one more question
Does this also apply to HvH lmfao

**Comment by zzurge** [comment]
mmmm 24/7 cadehug real

**Comment by shelteredfreak** [comment]
10/10 PR, merge instantly. Lore-friendly balance changes my beloved.

**Comment by breadhunt** [comment]
Idea that just popped into my head: Radiopack could have an "amplified" signal and so wearing it would bypass these restrictions. So that SLs, FCs etc. could still communicate with shipside (to some extent, and at a small cost).

Regardless this seems like a very interesting change, spedhmreg

**Comment by Pixelknightly** [comment]
This seems like a horrible idea, but honestly, I like it. Caves _are_ too comfy for marines and don't feel all that different from the outside even with all the advantages xenos tend to have inside caves. This makes it oppressive, which feels right, and makes it possible for a bunch of interesting scenarios to happen.

Please make radiopacks work flawlessly in caves though. This way those who are truly dedicated can still communicate with req and command while not having to rely on some weird messenger system while staying out of deep caves.

**Comment by Ragecrabbo** [comment]
This actually sounds really cool and im looking forward to seeing this ingame, however i have two questions.

How will certain maps like icy caves where the disks/nuke are in caves be changed to be less xeno favored after this change goes trough?

Will this affect gamemodes other than nukewar? As said earlier this would make HvH rather akward and on certain crash maps the canterbury will actually crash _inside_ the caves, meaning marines would have poor/no comms while in canter for the whole round.

**Comment by Losenis** [comment]
Me when xenos with no cas no OB and even with cavewide quickbuild still get skill issued

**Comment by Alexeivichi** [comment]
This is a horrible idea.

Merge it now.

**Comment by RipGrayson** [comment]
> How will certain maps like icy caves where the disks/nuke are in caves be changed to be less xeno favored after this change goes trough?

Adjustments will be made for certain maps if the changes affect things too harshly. I'm not sure of the specifics, but it might involve moving disk locations or changing certain areas of caves to be more marine friendly.

> Will this affect gamemodes other than nukewar? As said earlier this would make HvH rather akward and on certain crash maps the canterbury will actually crash _inside_ the caves, meaning marines would have poor/no comms while in canter for the whole round.

I'll probably adjust crash locations to remove the chances of a caves crash.

**Comment by RuneQuesteER** [comment]
> Comms works poorly in caves, and not at all in deep caves.

this is a terrible fucking PR, and I main xeno
speedmerge

**Comment by Xander3359** [comment]
Wow, this is awful.
Please merge.

**Comment by chizzy376** [comment]
Worst PR of this year hands down, speedmerge it right now

**Comment by Naaanii** [comment]
The yearly willpower PR is back.

**Comment by RikuTheKiller** [comment]
This is literally horrendous.
Merge it, now.

**Comment by KirovClassBattlecruiser** [comment]
Marines already get wiped their butt by the xenos everytime they decide to enter caves... 
I also think a sci fi headset would be able to communicate through caves lol
Solution would be to make caves narrower on the maps, not all the caves just some passages

**Comment by Coastfront** [comment]
"Get the fuck out of caves, why are you pushing?!" Trumpets the fc's voice, ever to hallowly echo unto a chamber of voicelessness. Never to find its meaning taking purchase in the hearts of those it longs to reach.

Marines already ignore command orders this is just going to further fuck over command players who are desperately trying to get john unga to leave caves before the marines throw a win because now they quite literally cant hear them

**Comment by Pixelknightly** [comment]
> "Get the fuck out of caves, why are you pushing?!" Trumpets the fc's voice, ever to hallowly echo unto a chamber of voicelessness. Never to find its meaning taking purchase in the hearts of those it longs to reach.
> 
> Marines already ignore command orders this is just going to further fuck over command players who are desperately trying to get john unga to leave caves before the marines throw a win because now they quite literally cant hear them

Announcement orders aren't going to be affected. And they're the most effective way to get marines to listen so this only really affects communication from caves to shipside, not shipside to caves. (Seriously, a ton of command orders get ignored not because marines don't feel like listening but because they're not paying attention to the radio. If command actually used announcement orders more often, they would listen more.)

Besides, if john unga decided he wants to go into the caves regardless of the fact that caves can make it impossible for him to yell for help, or for more ammo, or just point out where enemies are, he probably wasn't going to be of much help anyways.

This PR is (probably) a bad idea. But I wanna see it get TM'd anyways because there's a lot of possibilities here. Some of which are very good.

**Comment by breadhunt** [comment]
Yeah hard agree, split groups often won't listen to comms anyway. Gotta have a big voice guy listening out for comms there to keep everyone updated.

**Comment by DorfGuy1** [comment]
oh boy. peanut gallery here. This shit is gonna suck so hard. Not as hard as I respect this for making attempt at something being atmospheric and moody and stuff. Also for going balls to the wall about this. But really, give the marines some way to get some communications out. Like making radiopacks do radio things.

**Comment by David-Stormwell** [comment]
Up and down voted this for several reasons, the removal of supply beacon in caves seems fine, but comms nearly not working is a little wonky.
Why not implement a comms machine in which SLs need to maintain, or perhaps give the radio pack a duel reason for existing?

**Comment by Weedmaster77** [comment]
I think the idea of taking supplies being dropped into caves make sense and is fine but the whole taking radios away as a whole is seems overkill. I can see maybe reg marines radio not working in cave but think SLs and FC should have a boosted version so that they can still talk to shipside like someone said in earlier comment no OB and no CAS and now no radio or REQ your just asking for cade hugging and its just goin make marine less enjoyable with no real possibility of coordination in the moment only pre planning. Think that maybe make a REQ item that is like the Fire Support key that can be put into headsets to boost them to work in caves but make it like 120-150 points per one or something so that not everyone gets them but so that its still a option.

**Comment by Lumipharon** [comment]
People seem to forget there are multiple ways command/SL's can communicate without radio.

**Comment by isotope436** [comment]
there needs to be some sort of engi structure that gives radio in a small area or something cause this is provides nothing but annoyance and furthers the issue of zero command structure on lowpop

File: code/game/objects/items/radio/radio.dm [hunk 1]
```diff
@@ -291,6 +291,13 @@
 		signal.broadcast()
 		return
 
+	var/area/A = get_area(src)
+	if(istype(A))
```
Comment (Kapu1178) [inline code/game/objects/items/radio/radio.dm:295]:
```suggestion
	if(!isnull(A))
```

File: code/game/objects/items/radio/radio.dm [hunk 2]
```diff
@@ -339,6 +346,9 @@
 		var/turf/position = get_turf(src)
 		if(!position || !(position.z in levels))
 			return FALSE
+		var/area/A = get_area(src)
+		if(A && istype(A) && A.ceiling >= CEILING_DEEP_UNDERGROUND)
```
Comment (Kapu1178) [inline code/game/objects/items/radio/radio.dm:350]:
```suggestion
		var/area/A = get_area(src)
		if(A?.ceiling >= CEILING_DEEP_UNDERGROUND)
```

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

**Comment by ClosetedSkeletonGH** [comment]
Would be cool if restoring power to underground areas restores comms for that area

**Comment by ChocolateBickie** [comment]
I would suggest disabling minimap in deep caves too, as it's super easy to coordinate a push with just the minimap.

Unga tide for life!

**Comment by Overdrive659** [comment]
This is really cool, but radiopacks should have a radius around them that allows for comms. Or maybe squadcomms only and no general?? Its a very hard marine nerf.

**Pull request merged**